### PR TITLE
TST: refine ut and add 'convert_to_abspath' ut

### PR DESCRIPTION
**Pull request by aa452948257**
## What do these changes do?
refine ut and add 'convert_to_abspath' ut to cover 'read_xxx' function.

- [x] tests added / passed (if needed)
- [ ] Ensure all linting tests pass

### CODE REVIEW

File: .github/workflows/python.yaml [hunk 1]
```diff
@@ -165,7 +165,7 @@ jobs:
           pip install "xgboost_ray<0.1.14" "protobuf<4" "sqlalchemy<2"
         fi
 
-        pip install -e ".[dev,extra]"
```
Comment (qianduoduo0904) [inline .github/workflows/python.yaml:168]: Why do you remove the double quotes?

**Review by aresnow1 — APPROVED** [review]
LGTM

File: python/xorbits/_mars/dataframe/datasource/tests/test_datasource.py [hunk 2]
```diff
@@ -457,7 +457,8 @@ def test_read_csv():
             dtype=np.int64,
         )
         df.to_csv(file_path)
-        mdf = read_csv(file_path, index_col=0, chunk_bytes=10)
+        rel_path = os.path.relpath(file_path, start=os.curdir)
```
Comment (qianduoduo0904) [inline python/xorbits/_mars/dataframe/datasource/tests/test_datasource.py:460]: Is it possible to test both absolute path & rel path?

File: .github/workflows/python.yaml (hunk 1 above)
Comment (aa452948257) [reply .github/workflows/python.yaml:168]: when path is "s3", function "get_fs" requires "s3". So I rewrite this line "pip install -e .[dev,extra,aws]"

Comment (qianduoduo0904) [reply .github/workflows/python.yaml:168]: But you can keep the double quotes, I don't think it should be removed.

File: python/xorbits/_mars/dataframe/datasource/tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        local_relpath_lst = [local_path, local_path_2]
+        local_abspath_lst = convert_to_abspath(local_relpath_lst)
+        assert local_abspath_lst[0] == os.path.abspath(local_relpath_lst[0])
+        assert local_abspath_lst[1] == os.path.abspath(local_relpath_lst[1])
+        # test non-local/local path list convert
+        mix_relpath_lst = [s3_path, local_path]
+        mix_abspath_lst = convert_to_abspath(mix_relpath_lst)
+        assert mix_abspath_lst[0] == mix_relpath_lst[0]
+        assert mix_abspath_lst[1] == os.path.abspath(mix_relpath_lst[1])
+    finally:
```
Comment (ChengjieLi28) [inline python/xorbits/_mars/dataframe/datasource/tests/test_utils.py:40]: Why use try finally here?

**Review by UranusSeven — APPROVED** [review]
LGTM.

**Review by qianduoduo0904 — APPROVED** [review]
LGTM

**Review by qianduoduo0904 — APPROVED** [review]
LGTM

**Pull request merged**